Improve documentation of Unimod loading and highlight in ProForma

mobiusklein opened the pull request:
This PR implements some of the changes discussed in #82 but does not outright bundle the CV source file with the codebase. It adds documentation about how the CV can be loaded from disk instead of the internet.

It also re-structures the `pyteomics.proforma` documentation a bit and in need of a bit meditation I wrote a `fragments` method for the `Proforma` type.

mobiusklein commented:
I copied the docstrings because I cannot figure out how to tell `autodoc`'s `automodule` directive to just copy the module docstring and not list out the module members too. I couldn't make `:no-members:` or limiting it to only a subset of members. In order for the classes to be laid out in a way that isn't overwhelming to the reader.

I needed to disable `automodule` so I could intersperse them with prose to explain _why_ there are all these classes but the only thing they need to really worry about is the functional API or the `ProForma` object.

levitsky commented:
I feel like a lot of other documentation should be reworked for readability as well (outside of this PR, of course).

Do you think the module docstrings should be kept still? My concern is, the two versions will diverge; also I am not sure if anyone really benefits from the module-level documentation being there (unlike members' documentation, which often helps in interactive environments).

mobiusklein commented:
I removed the complicated module docstring content. I declined to put an actual URL to the docs in the docstring because I didn't know if you have persistent URL set up.

While I was at it I added a test for the new `fragments` method and made it 2.5x faster.

levitsky commented:
Thank you!
